core/space/services/services_fs.go

```diff
@@ -465,6 +465,61 @@ func (s *Space) AddItems(ctx context.Context, sourcePaths []string, targetPath s
 	return results, totalsRes, nil
 }
 
+func (s *Space) AddItemStream(ctx context.Context, fileName string, targetPath string, bucketName string) (domain.AddItemStreamResponse, error) {
+	err := s.waitForTextileInit(ctx)
+	if err != nil {
+		return domain.AddItemStreamResponse{}, err
+	}
+	path, err := os.Getwd()
+	fileLocalPath := filepath.Join(path, fileName)
+	f, err := os.Open(fileLocalPath)
+	if err != nil {
+		log.Error(fmt.Sprintf("error opening path %s", fileLocalPath), err)
+		return domain.AddItemStreamResponse{}, err
+	}
+	defer closeFile(f, fileLocalPath)
+
+	var targetPathBucket string
+	if targetPath == "" || targetPath == "/" {
+		targetPathBucket = fileName
+	} else {
+		targetPathBucket = targetPath + "/" + fileName
+	}
+	b, err := s.getBucketWithFallback(ctx, bucketName)
+	if err != nil {
+		return domain.AddItemStreamResponse{}, err
+	}
+	_, _, err2 := b.UploadFile(ctx, targetPathBucket, f)
+	if err2 != nil {
+		log.Error(fmt.Sprintf("error creating targetPath %s in bucket %s", targetPathBucket, b.Key()), err2)
+		return domain.AddItemStreamResponse{}, err
+	}
+	fi, err := f.Stat()
+	var fileSize int64 = 0
+	if err == nil {
+		fileSize = fi.Size()
+	}
+	return domain.AddItemStreamResponse{
+		Result:     "File Added",
+		TotalBytes: fileSize,
+	}, err
+}
+
+func closeFile(f *os.File, fileLocalPath string) {
+	fmt.Println("closing")
```

Comment:
Please use `log.Debug` instead of `fmt.Println`, although in this case I think we should just remove this line.

Reply:
removed